citar-embark.el

```diff
@@ -0,0 +1,130 @@
+;;; citar-embark.el --- Citar/Embark integration -*- lexical-binding: t; -*-
+;;
+;; Copyright (C) 2022 Bruce D'Arcus
+;;
+;; Author: Bruce D'Arcus <bdarcus@gmail.com>
+;; Maintainer: Bruce D'Arcus <bdarcus@gmail.com>
+;; Created: June 22, 2022
+;; Modified: June 22, 2022
+;; Version: 1.0
+;; Keywords: bib extensions
+;; Homepage: https://github.com/emacs-citar/citar-embark
+;; Package-Requires: ((emacs "27.2") (embark "0.17") (citar "0.9.5"))
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; Commentary:
+;;
+;;  Description
+;;
+;;; Code:
+
+(require 'citar)
+(require 'embark)
+
+;;;; Keymaps
+
+(defvar citar-embark-map (make-composed-keymap citar-map embark-general-map)
```

Comment:
Why not just move the maps here? They aren't useful outside Embark; are they?

Reply:
I bind `citar-map` to the prefix key `C-c b`, so I can open a bibliography file with `C-c b o`, insert a citation with `C-c b c`, etc. And it's nice to not have this prefix keymap cluttered with unnecessary keybinds from `embark-general-map`, so I figured it was better not to modify the original map in-place by setting its parent (which also makes `citar-embark-mode` more self-contained.

Reply:
I like the idea of the buffer actions being usable from a keybinding, or if possible also via a transient or hydra menu, without needing Embark.